### DISCUSSION TITLE
Discord Permissions and roles

### DISCUSSION THREAD

**Discussion by OBWANDO** [o3de/sig-operations#7] [General]
| Permission | ADMIN | MANAGER | MODERATOR | MOD-TRAIN | USER | Explanation |
| ---- | :-------: | :-------: | :-------: | :---: | :---: | ---- |
| **General Server Permissions** |
| View Channels              | X | X | X | X | X | Gives access to reading and seeing channels. |
| Manage channels            | X | X |   |   |   | Gives full access to the creation, deletion, reordering, and editing of channels or categories. Does not give access to server settings or any kind of permission settings for channels and categories. Does allow the managing of invites on channels directly. |
| Manage Roles               | X | X |   |   |   | Gives full access to the Roles tab of Server Settings allowing a user to change the settings of roles, create new roles and rearrange roles with the limitation of not being able to change the settings of their highest role or any roles above their highest role. Also allows the viewing of the Overview tab, and gives access to giving roles to members under the Members tab and anywhere else on the server with the limitation of only being able to give roles that are under their highest role. Furthermore, having the Manage Roles permission gives access to the Permissions tab of Category and Channel Settings. |
| Manage Emojis and Stickers | X | X |   |   |   | Allows the upload and deletion of Emojis in the Server Settings menu. Also gives access to viewing the Overview, Server Boost Status and Members tab in the server settings. |
| View Audit Log             | X | X | X | X |   | Allows viewing of the Audit Log tab under the Server Settings. Also gives access to seeing the Overview tab, Server Boost Status and the Members tab. |
| View Server Insights       | X | X | X |   |   | Allows the viewing of Server Insights under the Server Settings. Also gives access to seeing the Overview tab, Server Boost Status and the Members tab. |
| Manage Webhooks            | X | X |   |   |   | Gives access to the creation and deletion of Webhooks on Server Settings and Channel Settings under the Integration tab. Also gives access to viewing the Overview, Server Boost Status and Members tab in the server settings. |
| Manage Server              | X |   |   |   |   | Gives full access to changing the following Server Settings; Overview, Moderation, Integration, Widget and Server Template. It also allows viewing of the Server Boost Status and Members tab. Furthermore, roles with Manage Server can use the Invites tab to see all server invites and delete them. |
| **Membership Permissions** |
| Create Invite              | X | X | X | X | X | Allows for the creation of invite links to your server. This is the only permission that gives this ability, if this is turned off a user can not invite people to your server. |
| Change Nickname            | X | X | X | X | X | Lets a user change their nickname on your server to something custom from their username. |
| Manage Nicknames           | X | X | X |   |   | Works the same as Change Nickname with the addition of having the ability to change other users’ nicknames. Also gives access to viewing the Overview, Server Boost Status and Members tab in Server Settings. |
| Kick Members               | X | X | X |   |   | Allows a user to kick other members from the server. Also gives access to using the Prune function under the Members tab of Server Settings, but this permission does not give access to Server Settings, so another permission has to be in place that allows the user to see the Server Settings such as View Audit Log. |
| Ban Members                | X | X |   |   |   | Allows the banning and unbanning of users. Gives access to viewing the Overview, Server Boost Status, Members and Bans tab under Server Settings. |
| **Text Channel Permissions** |
| Send Messages              | X | X | X | X | X |Allows for sending messages. |
| Public Threads               | X | X | X | X | X | When the "Send Messages" permission is also enabled, allows members to both create and talk in threads. When the "Send Message" is disabled, only allows members to talk in existing threads, but not create new threads. |
| Private Threads               | X | X | X | X |   | When the "Send Messages" permission is also enabled, allows members to both create and talk in private threads. When the "Send Messages" is disabled, only allows members to talk in existing private threads, but not create new private threads. |
| Embed Links                | X | X | X | X | X | If enabled when a user sends a link Discord will automatically embed a preview of the link. For example, sending a YouTube video link will create a YouTube video player embed to play the video straight in the application. Bots usually needs this to send embeds. |
| Attach Files               | X | X | X | X | X | Lets a user send files and images from their computer/mobile device. |
| Add Reactions              | X | X | X | X | X | Lets a user add new reactions to messages. This permission is not needed for reactions already added to a message. |
| Use External Emoji         | X | X | X | X | X | Allows users to use emojis from other servers in messages, reactions and custom status messages. Requires Discord Nitro usually, but this also applies to emojis obtained through other means like Twitch subscriptions and Global Emotes. |
| Mention @everyone, @here, and All Roles | X | X | X |   |   | Allows for the mentioning of the everyone role and the @here mention. It also allows the mentioning of any role in the server, overwriting the “Allow anyone to @mention this role” setting. This is best left off on normal user roles as it can be used to spam people with notifications. |
| Manage Messages            | X | X | X |   |   |
| Manage Threads            | X | X | X |   |   | Allows members to rename, delete, archive/unarchive, and turn on slow mode for threads. They can also view private threads. |
| Read Message History       | X | X | X | X | X | Allows for the reading of message history. When this permission is denied, a user that exits and reloads Discord or refreshes it with CTRL+R will not see the content sent in channels from before the the reload. Even their own messages will not show. As long as the application is running anything sent during that time will still be available to read. |
| Send Text-To-Speech Messages | X | X | X | X |   | Lets a user send a text-to-speech message using the /tts text command. A user has to be viewing the channel to have the audio play otherwise it will just look like a normal message. |
| Use Slash Commands         | X | X | X | X | X | Allows members of your server to use slash commands in text channels. This can be controlled by server or by channel. The biggest usage for this is controlling if people can use the slash commands of bots in any chat, keep in mind though they won’t be able to use Discord’s integrated slash commands if you turn this off. | 
| **Voice Channel Permissions** |
| Connect                    | X | X | X | X | X | Allows a user to connect to a voice channel. |
| Speak                      | X | X | X | X | X | Gives access to speaking in a voice channel. |
| Video                      | X | X | X | X | X | Allows the streaming of video such as Going Live with a game, screen sharing and video calls. |
| Use Voice Activity         | X | X | X | X | X | Allows for the use of voice activity instead of being forced to use Push-To-Talk. |
| Priority Speaker           | X | X | X | X |   | Users with this permission will sound louder than users without this permission. Priority Speaker is activated by using the Push to Talk (Priority) keybind. |
| Mute Members               | X | X | X | X |   | Allows the server muting of users in voice channels. Does not affect text. |
| Deafen Members             | X | X | X | X |   | Allows for the server deafening of users in voice channels. Does not effect text. |
| Move Members               | X | X | X | X |   | Lets a user move other users from one voice channel to another or disconnect a user from the voice channel.|
| **Stage Channel Permissions** |
| Request to Speak           | X | X | X | X | X | Allows users to request to speak in a Stage channel. Moderators have to allow or deny the request manually while in the channel. |
| **Advanced Permissions** |
| Administrator              |   |   |   |   |   | Gives full access to the server and all its settings, also overwrites all permissions. This should only be given to roles that are intended to have the same control over the server as the server owner. | 

Channel permissions: 
Information category: Admin and Manager permissions
Administrator permission provided only to members that are also in private channels that are for Governing board, TSC, and release as they contain sensitive information.

**Comment by sptramer** [comment]
Only official Discord documentation I could find on roles and moderation: https://discord.com/moderation/1500000176222-201:-Permissions-on-Discord

It looks like all of these are covered here, but I couldn't find a full and comprehensive list of user capabilities and which ones are toggleable by mods. Seems reasonable.

**Comment by JustFinchy** [comment]
I would suggest to only allow Managers and admins to Ban users. Mods being able to kick, and mute members, or even a temp ban is enough, but a Perm ban should only be done by managers and admins. 

Though usually it also is like 1 in every few hundred users that this is needed. 

Other than that, looks good. 


Might want to have a "Jr Mod" or a "Mod in Training" that would be a limited Mod perms for users in training that have yet to have full perms a Mod would have. If that makes sense.

**Reply by OBWANDO** [reply]
Updated with the new Mod Train role

**Reply by JustFinchy** [reply]
Thanks, :)

Saw one of the Mods in Training request a non-mod to see whats different.

**Reply by OBWANDO** [reply]
Updated with standard user permissions

**Comment by rainbj** [comment]
Hi @OBWANDO and @aFinchy. Just helping to manage and maintain the UX side of things. Please let me know if I need to be apart of any conversations from your end. Many thanks. Joshua

**Comment by JustFinchy** [comment]
There is a new Role "Events" that needs to be added. Would be nice to test how events work, and then role out to verified users for sigs to setup events for meetings
